crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -678,6 +687,51 @@ impl FileSystem for DiskFileSystem {
 
         let _lock = self.mutex_map.lock(full_path.clone()).await;
 
+        // Make sure that there is only one authorative writer at a time.
+        let conflict = {
+            let own_invalidator = get_invalidator();
+            let mut authorative_write_map = self.authorative_write_map.lock().unwrap();
+            let inner_map = authorative_write_map.entry(full_path.clone()).or_default();
+            let mut new_map = AutoMap::new();
+            let mut has_own = false;
+            // Remove all inactive invalidators.
+            for (invalidator, state) in take(inner_map).into_iter() {
+                if invalidator == own_invalidator {
+                    has_own = true;
+                    new_map.insert(invalidator, state);
+                } else if invalidator.is_active() {
+                    new_map.insert(invalidator, state);
+                } else {
+                    invalidator.invalidate();
```

Comment:
Do we need to invalidate? If this is inactive, couldn't we just drop it?

Reply:
It might become active later and we want to recompute it in this case to make the conflict check again.